src/engraving/dom/property.cpp

```diff
@@ -501,6 +501,17 @@ bool propertyLink(Pid id)
     return propertyList[int(id)].link;
 }
 
+//---------------------------------------------------------
+//   propertyLinkSameScore
+//---------------------------------------------------------
+
+bool propertyLinkSameScore(Pid id)
+{
+    assert(id < Pid::END);
+    std::array<Pid, 3> doNotLinkSameScore { Pid::STAFF_BARLINE_SPAN, Pid::STAFF_BARLINE_SPAN_FROM, Pid::STAFF_BARLINE_SPAN_TO };
```

Comment:
This should be static, otherwise it gets instantiated at every call of the function. Also, an array is not the best data structure for this because searching an array (as in muse::contains) scales linearly with the size. A better option would be an std::set.

Though to be honest for this kinds of logic which require a stricly binary choice between a group of cases and a default behaviour my preferred approach (which I'm quite sure is also the fastest) is just a lame switch statement, such as
```
switch (id) {
    case Pid::STAFF_BARLINE_SPAN:
    case Pid::STAFF_BARLINE_SPAN_FROM:
    case Pid::STAFF_BARLINE_SPAN_TO:
        return false;
    default:
        return true;
}
```